### ISSUE TITLE
gridcontainer hint/tooltip disturb the layout

### ISSUE THREAD

**Issue by astrale-sharp** [godotengine/godot#46976]
**Godot version:**

3.2.3.stable

**OS/device including version:**

```
❯ screenfetch
/usr/bin/screenfetch: ligne 2549: gconftool-2 : commande introuvable
                                       astrale@Sharp
 MMMMMMMMMMMMMMMMMMMMMMMMMmds+.        OS: Linuxmint 20.1 ulyssa
 MMm----::-://////////////oymNMd+`     Kernel: x86_64 Linux 5.4.0-66-generic
 MMd      /++                -sNMd:    Uptime: 11h 41m
 MMNso/`  dMM    `.::-. .-::.` .hMN:   Packages: 4211
 ddddMMh  dMM   :hNMNMNhNMNMNh: `NMm   Shell: fish 3.1.0
     NMm  dMM  .NMN/-+MMM+-/NMN` dMM   Resolution: 1920x1080
     NMm  dMM  -MMm  `MMM   dMM. dMM   DE: Cinnamon 4.8.6
     NMm  dMM  -MMm  `MMM   dMM. dMM   WM: Metacity
     NMm  dMM  .mmd  `mmm   yMM. dMM   GTK Theme: Mint-Y-Dark-Red [GTK2/3]
     NMm  dMM`  ..`   ...   ydm. dMM   Icon Theme: Mint-Y-Dark-Red
     hMM- +MMd/-------...-:sdds  dMM   Font: Ubuntu Condensed, 24
     -NMm- :hNMNNNmdddddddddy/`  dMM   Disk: 841G / 1,1T (87%)
      -dMNs-``-::::-------.``    dMM   CPU: Intel Core i3-2330M @ 4x 2.2GHz [74.0°C]
       `/dMNmy+/:-------------:/yMMM   GPU: GeForce GT 630M
          ./ydNMMMMMMMMMMMMMMMMMMMMM   RAM: 2732MiB / 3844MiB
             \.MMMMMMMMMMMMMMMMMMM    
                                      
```

**Issue description:**

Hi!
When you're using a GridContainer with some hint you get a jittery result like that : 
![grid_container](https://user-images.githubusercontent.com/53686698/111042209-86d5e100-843c-11eb-8bfc-954514c57292.gif)

When you're using a HBoxcontainer with some tooltip you get a nice message on top that doesn't disturb the alignment like this:
![hbox_container](https://user-images.githubusercontent.com/53686698/111042227-9e14ce80-843c-11eb-8189-1b775901c4ea.gif)

**Steps to reproduce:**
add a GridContainer with some tool tip to a scene.

**Comment by Calinou** [comment]
@astrale-sharp Please upload a [minimal reproduction project](https://github.com/godotengine/godot/blob/master/CONTRIBUTING.md#provide-a-simple-example-project) to make this easier to troubleshoot.

**Comment by kuhnchris** [comment]
![image](https://user-images.githubusercontent.com/761911/111051403-079fd780-8453-11eb-8c13-d897932d7544.png)
![image](https://user-images.githubusercontent.com/761911/111051409-0ff81280-8453-11eb-8658-b4f2611700a4.png)
The issue gets more complicated the more items you add to the grid:
![image](https://user-images.githubusercontent.com/761911/111051417-230ae280-8453-11eb-9dbc-c5a1fad56132.png)
I didn't get to explode it like OP did tho. Attached is a project with the issue at hand.
It is, however, affected by the length of the tool tip:
![image](https://user-images.githubusercontent.com/761911/111051445-52215400-8453-11eb-8b56-47f6c1a4048e.png)
(see the button)

[me-46976.zip](https://github.com/godotengine/godot/files/6135708/me-46976.zip)

**Comment by kuhnchris** [comment]
On further investigation: https://github.com/godotengine/godot/blob/942f5f0cb282cbc2b3fb766e3c7614e82b98dd9b/scene/main/viewport.cpp#L1579-L1587
I assume this may be caused due to the TooltipPanel being parented to the GridContainer (and HBoxContainer).
The HBoxContainer _resort() function is very different to the grid_container one, tho I would fail to see where this may cause an issue, but i'm just a godot programmer rookie, so maybe the seniors may know a bit more.
But just by the look of it, wouldn't it make more sense to parent the tooltip to the scene_root instead of the "tooltip owner"? Because I assume that this may be the main issue of this, as the aforementioned parenting happens directly on the node tree (if you investigate via the Remote tab in Godot):
![image](https://user-images.githubusercontent.com/761911/111052105-716eb000-8458-11eb-8ebe-d8df4079dcf0.png)
(HBoxContainer, works)
![image](https://user-images.githubusercontent.com/761911/111052112-821f2600-8458-11eb-9cf0-6eefc4d6b520.png)
(GridContainer, explodes)

**Comment by KoBeWi** [comment]
This is fixed in 4.0, but the fix can't be backported (because it happened with DisplayServer rewrite).

**Comment by astrale-sharp** [comment]
Damn i finished my minimal reproduction project before reading this was fixed ~~
I dont upload then, right?

If this is fixed, i guess we can close the issue

**Comment by Calinou** [comment]
To solve this, shouldn't tooltips be added high up in the scene tree?

**Comment by KoBeWi** [comment]
Yeah, we could add them directly under viewport. But maybe there was a reason why it wasn't done in the first place.
